pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -424,7 +422,7 @@ func checkAndKeepInitContainers(pod *v1.Pod, podStatus *kubecontainer.PodStatus,
 }
 
 // computePodContainerChanges checks whether the pod spec has changed and returns the changes if true.
-func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podStatus *kubecontainer.PodStatus) podContainerSpecChanges {
+func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podStatus *kubecontainer.PodStatus, existingStatus *v1.PodStatus) podContainerSpecChanges {
```

Comment:
this is confusing as well.

existingStatus == prevObservedStatus
podStatus == observedStatus